src/communication/index.ts

```diff
@@ -1,108 +1,70 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, Response } from '../types';
+import { MessageFormatter } from './messageFormatter';
 
-class InterfaceCommunicator implements Communicator {
-  private allowedOrigins: RegExp[] = [];
+// eslint-disable-next-line
+type Callback = (response: any) => void;
+
+class PostMessageCommunicator implements Communicator {
+  private allowedOrigins: RegExp[] | null = null;
```

Comment:
I think this should be blocked origins, right?

Reply:
hmm  why?
```
    if (Array.isArray(this.allowedOrigins)) {
      validOrigin = this.allowedOrigins.find((regExp) => regExp.test(origin)) === undefined;
    }
```

if they're provided we check the origin to match one of the reg expressions

Reply:
I will close this PR because there's some black sorcery going on with the ESLint action. It works fine for other projects tho. I will try to open a new one and see if it will fix